### PR TITLE
Prefab core fixes

### LINKED ISSUE #94: Installation of nodejs from prefab failed
#### Installation information

- jumpscale version: master
- operating system: linux

#### Detailed description
I tried to install nodejs using prefab, and it failed. 
Fun fact. after I get the error. the `/tmp` doesn't exist anymore. So something in the process deletes `/tmp` !!

#### Steps to reproduce
execute `j.tools.prefab.local.runtimes.nodejs.install()`
 from the jshell.

#### Relevant stacktraces
```
FileNotFoundError                         Traceback (most recent call last)
/usr/local/bin/js9 in <module>()
----> 1 j.tools.prefab.local.runtimes.nodejs.install()

/opt/code/github/jumpscale/prefab9/modules/runtimes/PrefabNodeJS.py in install(self, reset)
     96         cdest = self.prefab.core.file_download(
     97             url, expand=True, overwrite=False, to="$TMPDIR")
---> 98 
     99         self.core.run("rm -rf /opt/node;mv %s /opt/node" % (cdest))
    100         self.core.run("rm /usr/local/bin/node;ln -s /opt/node/bin/node /usr/local/bin/node")

/opt/code/github/jumpscale/core9/JumpScale9/tools/bash/BashFactory.py in profileDefault(self)
    281         if self._profileDefault is None:
    282             path = "~/.bash_profile"
--> 283             self._profileDefault = self.profileGet(path)
    284         return self._profileDefault
    285 

/opt/code/github/jumpscale/core9/JumpScale9/tools/bash/BashFactory.py in profileGet(self, path)
    263 
    264     def profileGet(self, path="~/.profile_js"):
--> 265         path = path.replace("~", self.home)
    266         if not self.executor.exists(path):
    267             self.executor.file_write(path, "")

/opt/code/github/jumpscale/core9/JumpScale9/tools/bash/BashFactory.py in home(self)
    241     @property
    242     def home(self):
--> 243         return self.executor.env.get("HOME") or self.executor.dir_paths["HOMEDIR"]
    244 
    245     def cmdGetPath(self, cmd, die=True):

/opt/code/github/jumpscale/core9/JumpScale9/tools/executor/ExecutorBase.py in env(self)
     53     @property
     54     def env(self):
---> 55         return self.stateOnSystem["env"]
     56 
     57     def docheckok(self, cmd, out):

/opt/code/github/jumpscale/core9/JumpScale9/tools/executor/ExecutorBase.py in stateOnSystem(self)
    205             """
    206             C = j.data.text.strip(C)
--> 207             rc, out, err = self.execute(C, showout=False, hide=True)
    208 
    209             res = {}

/opt/code/github/jumpscale/core9/JumpScale9/tools/executor/ExecutorLocal.py in execute(self, cmds, die, checkok, showout, outputStderr, timeout, env, hide)
     43 
     44         rc, out, err = j.sal.process.execute(cmds2, die=die, showout=showout,
---> 45                                              outputStderr=outputStderr, timeout=timeout)
     46 
     47         if checkok:

/opt/code/github/jumpscale/core9/JumpScale9/sal/process/SystemProcess.py in execute(self, command, showout, outputStderr, useShell, log, cwd, timeout, errors, ok, captureout, die, async)
     61 
     62         if "\n" in command:
---> 63             path = j.sal.fs.getTmpFilePath()
     64             self.logger.debug("execbash:\n'''%s\n%s'''\n" % (path, command))
     65             if die:

/opt/code/github/jumpscale/core9/JumpScale9/fs/SystemFS.py in getTmpFilePath(self, cygwin)
   1378         # return tempfile.mktemp())
   1379         tmpdir = j.dirs.TMPDIR
-> 1380         fd, path = tempfile.mkstemp(dir=tmpdir)
   1381         try:
   1382             real_fd = os.fdopen(fd)

/usr/lib/python3.5/tempfile.py in mkstemp(suffix, prefix, dir, text)
    479         flags = _bin_openflags
    480 
--> 481     return _mkstemp_inner(dir, prefix, suffix, flags, output_type)
    482 
    483 

/usr/lib/python3.5/tempfile.py in _mkstemp_inner(dir, pre, suf, flags, output_type)
    397         file = _os.path.join(dir, pre + name + suf)
    398         try:
--> 399             fd = _os.open(file, flags, 0o600)
    400         except FileExistsError:
    401             continue    # try again

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmprhkh4fud'
```

### PR DESCRIPTION
**Pull request by zaibon**
#### What this PR resolves:
fixes #94 

#### Changes proposed in this PR:

-  fix joinpath implementation
- support .tar.gz extension when decompression archives

### CODE REVIEW

**Comment by zaibon** [comment]
Rebased my branch, no more conflict now.

**Comment by zaibon** [comment]
> Jo De Boeck
os.path.join works for the local executor so it uses os.sep of the local environment
prefab needs to be aware of the arch of its host where it is executing on
so os..path.join can not be used

So I'll revert this

File: JumpScale9Prefab/PrefabCore.py [hunk 1]
```diff
@@ -1063,13 +1065,8 @@ def file_md5(self, location):
...
     def joinpaths(self, *args):
         path = ""
-        seperator = "\\"
-        if self.isMac or self.isUbuntu or self.isArch:
-            seperator = "/"
-        for arg in args:
-            path += "%s%s" % (seperator, arg)
-        path = self.replace(path)
-        return path
+        path = os.path.join(path, *args)
```
Comment (xmonader) [inline JumpScale9Prefab/PrefabCore.py]: os.path.join works fine for local machines. this is executing through prefab so you don't really know what os u have at this moment

Comment (zaibon) [reply JumpScale9Prefab/PrefabCore.py]: yep you're right. Pushed a new version that handle this and still improve a bit the logic.
It maps the behavior of `os.path.join` but check first which separator to use

**Pull request closed without merging**